lib/deimos/utils/db_poller.rb

```diff
@@ -143,15 +150,50 @@ def fetch_results(time_from, time_to)
 
       # @param batch [Array<ActiveRecord::Base>]
       # @return [void]
-      def process_batch(batch)
+      def process_batch_with_span(batch)
+        retries = 0
+        begin
+          span = Deimos.config.tracer&.start(
+            'deimos-db-poller',
+            resource: @producer.class.name.gsub('::', '-')
+          )
+          process_batch(batch)
+          Deimos.config.tracer&.finish(span)
+        rescue Kafka::Error => e # keep trying till it fixes itself
+          Deimos.config.logger.error("Error publishing through DB Poller: #{e.message}")
+          sleep(0.5)
+          retry
+        rescue StandardError => e
+          if retries < @config.retries
+            retries += 1
+            sleep(0.5)
+            retry
+          else
+            Deimos.config.tracer&.set_error(span, e)
```

Comment:
I was able to see the retry and tracer working with local setup. 
> Mock span 'deimos-db-poller' set an error: at .item_id expected type string, got null

Would you find it useful to log the error here in case they don't use datadog tracer?